### PR TITLE
fix: LEAP-861: Resolve CUDA runtime error issue

### PR DESCRIPTION
**Pull request by niklub**
Fixed runtime error with executing ML backend in CUDA environment with gunicorn:

```
RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method
```

More details in [the issue](https://github.com/HumanSignal/label-studio-ml-backend/issues/79)

### CODE REVIEW

File: label_studio_ml/examples/segment_anything_model/start.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 # Execute the gunicorn command
-exec gunicorn --preload --bind :${PORT:-9090} --workers 1 --threads 8 --timeout 0 _wsgi:app
+exec gunicorn --bind :${PORT:-9090} --workers 1 --threads 8 --timeout 0 _wsgi:app
```
Comment (hakan458) [inline label_studio_ml/examples/segment_anything_model/start.sh:4]: Why was `--preload` removed?

Comment (niklub) [reply label_studio_ml/examples/segment_anything_model/start.sh:4]: https://stackoverflow.com/questions/72779926/gunicorn-cuda-cannot-re-initialize-cuda-in-forked-subprocess
https://github.com/pytorch/pytorch/issues/116543

File: label_studio_ml/examples/segment_anything_model/README.md [hunk 2]
```diff
@@ -14,6 +14,21 @@ To start the server with lightweight mobile version of SAM, run the following co
 docker-compose up
 ```
 
+### GPU Support
+
+By default, the docker-compose file runs the model on the CPU. If you have a GPU, you can enable it by adding the following lines in `docker-compose.yml`:
+
+```yaml
+    environment:
+      - NVIDIA_VISIBLE_DEVICES=all
```
Comment (mpesavento) [inline label_studio_ml/examples/segment_anything_model/README.md:23]: Is this line suggested for other yaml files? it doesnt appear in the `segment_anything_model` example.

File: label_studio_ml/examples/segment_anything_model/sam_predictor.py [hunk 3]
```diff
@@ -2,16 +2,18 @@
...
 
 logger = logging.getLogger(__name__)
+_MODELS_DIR = pathlib.Path(__file__).parent / "models"
 
-VITH_CHECKPOINT = os.environ.get("VITH_CHECKPOINT", "sam_vit_h_4b8939.pth")
-ONNX_CHECKPOINT = os.environ.get("ONNX_CHECKPOINT", "sam_onnx_quantized_example.onnx")
-MOBILESAM_CHECKPOINT = os.environ.get("MOBILESAM_CHECKPOINT", "mobile_sam.pt")
+VITH_CHECKPOINT = os.environ.get("VITH_CHECKPOINT", _MODELS_DIR / "sam_vit_h_4b8939.pth")
+ONNX_CHECKPOINT = os.environ.get("ONNX_CHECKPOINT", _MODELS_DIR / "sam_onnx_quantized_example.onnx")
+MOBILESAM_CHECKPOINT = os.environ.get("MOBILESAM_CHECKPOINT", _MODELS_DIR / "mobile_sam.pt")
```
Comment (mpesavento) [inline label_studio_ml/examples/segment_anything_model/sam_predictor.py:16]: Thanks for making this explicit!

File: label_studio_ml/wsgi.py [hunk 4]
```diff
@@ -4,6 +4,16 @@
...
+try:
+    import torch.multiprocessing as mp
+    try:
+        # avoid "cannot reinit CUDA in forked process" error in loading cuda?
+        mp.set_start_method('spawn')
+    except RuntimeError:
+        pass
+except ImportError:
+    import multiprocessing as mp
+
```
Comment (mpesavento) [inline label_studio_ml/wsgi.py:16]: Why does this work here and not if I put the same commands into the example `_wsgi.py`?

**Review by mpesavento — APPROVED** [review]
Thank you for this fix and unblocking my workflow!

**Comment by mpesavento** [comment]
It looks like all of the test suites are running the SAM test, which is failing for unknown reasons:
```
ERROR label_studio_ml/examples/segment_anything_model/test_predict.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
```

File: label_studio_ml/wsgi.py (hunk 4 above)
Comment (niklub) [reply label_studio_ml/wsgi.py]: It actually does not work with gunicorn. The only solution is to remove `--preload`

File: label_studio_ml/examples/segment_anything_model/README.md (hunk 2 above)
Comment (niklub) [reply label_studio_ml/examples/segment_anything_model/README.md:23]: added to the docker-compose as well

File: label_studio_ml/examples/segment_anything_model/docker-compose.yml [hunk 5]
```diff
@@ -17,14 +19,24 @@ services:
...
       # Add these variables if you want to access the images stored in Label Studio, host should start with http:// or https://
       - LABEL_STUDIO_HOST=  
       - LABEL_STUDIO_ACCESS_TOKEN=
+      # Enable this to use the GPU
+      # - NVIDIA_VISIBLE_DEVICES=all
+      # specify the number of workers and threads for the model server
+      - WORKERS=1
+      - THREADS=8
+      # specify the model directory (likely you don't need to change this)
+      - MODEL_DIR=/data/models
```
Comment (mpesavento) [inline label_studio_ml/examples/segment_anything_model/docker-compose.yml:39]: nit: this should be `MODELS_DIR` to match the env var in download_models.sh

**Pull request merged**